### ISSUE TITLE
[eslint-plugin-lit-a11y]: implement no-noninteractive-tabindex

### ISSUE THREAD

**Issue by bennypowers** [open-wc/open-wc#1929]
see: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-tabindex.md

# no-noninteractive-tabindex

Tab key navigation should be limited to elements on the page that can be interacted with. Thus it is not necessary to add a tabindex to items in an unordered list, for example, to make them navigable through assistive technology. These applications already afford page traversal mechanisms based on the HTML of the page. Generally, we should try to reduce the size of the page's tab ring rather than increasing it.

## Rule details

The recommended options for this rule allow `tabindex` on elements with the noninteractive `tabpanel` role. Adding `tabIndex` to a tabpanel is a recommended practice in some instances.

```javascript
'lit-a11y/no-noninteractive-tabindex': ['error',
  {
    tags: [],
    roles: ['tabpanel'],
  },
]
```

### Succeed
```js
html`
  <div></div>
  <custom-button tabindex="0"></custom-button>
  <custom-button tabindex=${0}></custom-button>
  <button></button>
  <button tabIndex="0"></button>
  <button tabIndex=${0}></button>
  <div tabindex="-1"></div>
  <div role="button" tabindex="0"></div>
  <div role="article" tabindex="-1"></div>
  <article tabindex="-1"></article>
`;
```

### Fail
```js
html`
  <div tabindex="0"></div>
  <div role="article" tabindex="0"></div>
  <article tabindex="0"></article>
  <article tabindex={0}></article>
`;
```

## Accessibility guidelines
- [WCAG 2.1.1](https://www.w3.org/WAI/WCAG21/Understanding/keyboard)

### Resources
- [Fundamental Keyboard Navigation Conventions](https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_generalnav)
